RecoLocalTracker/SiPixelRecHits/interface/pixelCPEforDevice.h

```diff
@@ -338,6 +345,51 @@ namespace pixelCPEforDevice {
     }
   }
 
+  template <typename TrackerTraits>
+  constexpr inline void irradiationBiasCorrection(CommonParams const& __restrict__ comParams,
+                                                  DetParams const& __restrict__ detParams,
+                                                  ClusParams& cp,
+                                                  uint32_t ic) {
+    if (!comParams.irradiationBiasCorrection_) {
+      return;
+    }
+
+    float ibc_x = 0;
+    float ibc_y = 0;
+
+    // in detParams qBins are reversed bin0 -> smallest charge, bin4-> largest charge
+    // whereas in CondFormats/SiPixelTransient/src/SiPixelGenError.cc it is the opposite
+    // so we reverse the bin here -> kGenErrorQBins - 1 - bin
+    int bin = 0;
+    bin = kGenErrorQBins - 1 - cp.status[ic].qBin;
+
+    if (cp.status[ic].isOneX) {  // size=1
+      //for size = 1, the Lorentz shift is already accounted by the irradiation correction
+      ibc_x = ibc_x - detParams.shiftX;
+      if (cp.status[ic].isBigX) {
+        ibc_x -= detParams.dx2;
+      } else {
+        ibc_x -= detParams.dx1;
+      }
+    } else {  // size>1
+      ibc_x -= detParams.deltax[bin];
```

Comment:
Also this, could be
```suggestion
      ibc_x = -detParams.deltax[bin];
```
?

Reply:
This and above suggestions were implemented in the next commit.